src/goto-symex/symex_target.h

```diff
@@ -63,21 +69,46 @@ class symex_targett
     GUARD,
   };
 
-  // read event
+  /// Reading from a shared variable \c ssa_rhs (which will thus become a
+  /// left-hand side as well): we effectively assign the value stored in
+  /// \c ssa_rhs by another thread to \c ssa_rhs in the memory scope of this
+  /// thread.
+  /// \param guard: condition to take this read event
+  /// \param ssa_rhs: variable to be read from
+  /// \param atomic_section_id: ID of the atomic section in which this read
+  /// takes place
+  /// \param source: pointer to location in the input GOTO program of this read
   virtual void shared_read(
     const exprt &guard,
     const ssa_exprt &ssa_rhs,
     unsigned atomic_section_id,
     const sourcet &source)=0;
 
-  // write event
+  /// Writing to a shared variable \c ssa_rhs (which will thus become a
+  /// left-hand side as well): we effectively assign a value from this thread
+  /// to be visible by other threads.
+  /// \param guard: condition to take this write event
```

Comment:
:strawberry:

Reply:
Hopefully all the strawberries should be handled.